### PR TITLE
feat(EmptyState): Convert empty state to TypeScript

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Refactored empty state to TypeScript.

Fixes https://github.com/patternfly/patternfly-react/issues/1991.

### CODE REVIEW

**Comment by nicolethoen** [comment]
@rebeccaalpert can you also add integration tests for this component as outlined on this [README](https://github.com/patternfly/patternfly-react/tree/master/packages/patternfly-4/react-integration)?

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  small: styles.modifiers.sm,
+  full: null
+}
+
+export const EmptyState: React.FunctionComponent<EmptyStateProps> = ({
+  children,
+  className = '',
+  variant = EmptyStateVariant.large,
+  ...props
+}) => {
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx]: Change to `}: EmptyStateProps) => {`

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const maxWidthModifier = variant && maxWidthModifiers[variant];
+  return (
+    <div className={css(styles.emptyState, maxWidthModifier && maxWidthModifier, className)} {...props}>
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx]: I know you didn't write this logic, but please change `maxWidthModifier && maxWidthModifier` to just `maxWidthModifier`. Anything `&&` with itself is itself.

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateBody.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  children?: React.ReactNode;
+  /** Additional classes added to the EmptyState */
+  className?: string;
+}
+
+export const EmptyStateBody: React.FunctionComponent<EmptyStateBodyProps> = ({
+  children = null,
+  className = '',
+  ...props
+}) => (
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateBody.tsx]: Change to `}: EmptyStateBodyProps) => (`

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateBody.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+export interface EmptyStateBodyProps extends React.HTMLProps<HTMLParagraphElement> {
+  /** Content rendered inside the EmptyState */
+  children?: React.ReactNode;
+  /** Additional classes added to the EmptyState */
+  className?: string;
+}
+
+export const EmptyStateBody: React.FunctionComponent<EmptyStateBodyProps> = ({
+  children = null,
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateBody.tsx]: Remove `= null`, the JS doesn't set a default (```EmptyStateBody.defaultProps = {
  className: ''
};```)

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  color?: string;
+  size?: IconSize;
+  title?: string;
+}
+
+export interface EmptyStateIconProps extends IconProps {
+  /** Additional classes added to the EmptyState */
+  className?: string;
+  /** Icon component to be rendered inside the EmptyState */
+  icon: string | React.FunctionComponent<IconProps>;
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateIcon.tsx]: I think `string | React.ReactType<IconProps>` might be a more appropriate type if it works.

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+export const EmptyStateIcon: React.FunctionComponent<EmptyStateIconProps> = ({
+  className = '',
+  icon: IconComponent,
+  ...props
+}) => (
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateIcon.tsx]: Change to `}: EmptyStateIconProps) => (`

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateSecondaryActions.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  /** Content rendered inside the EmptyState */
+  children?: React.ReactNode;
+  /** Additional classes added to the EmptyState */
+  className?: string;
+}
+export const EmptyStateSecondaryActions: React.FunctionComponent<EmptyStateSecondaryActionsProps> = ({
+  children = null,
+  className = '',
+  ...props
+}) => (
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateSecondaryActions.tsx]: Change to `}: EmptyStateSecondaryActionsProps) => (`

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  full = 'full'
+}
+
+export interface EmptyStateProps extends React.HTMLProps<HTMLDivElement> {
+  /** Additional classes added to the EmptyState */
+  className?: string;
+  /** Content rendered inside the EmptyState */
+  children: React.ReactNode;
+  /** Modifies EmptyState max-width */
+  variant?: EmptyStateVariant;
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx]: This should be updated to use the strings.

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+export enum IconSize {
+  sm = 'sm',
+  md = 'md',
+  lg ='lg',
+  xl ='xl'
+}
+
+export interface IconProps extends Omit<React.HTMLProps<SVGElement>, 'size'> {
+  color?: string;
+  size?: IconSize;
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyStateIcon.tsx]: We should use strings here.

**Comment by rebeccaalpert** [comment]
Added test/demo and addressed PR feedback.

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/EmptyStateDemo/EmptyStateDemo.tsx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+     <React.Fragment>
+       <EmptyState children={this.myLargeEmptyStateProps.children} />
+       <EmptyState children={this.mySmallEmptyStateProps.children} variant={this.mySmallEmptyStateProps.variant} />
+       <EmptyState children={this.myFullEmptyStateProps.children} variant={this.myFullEmptyStateProps.variant} />
+     </React.Fragment>
+   );
+ }
+}
+
+export default EmptyStateDemo;
```
Comment (redallen) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/EmptyStateDemo/EmptyStateDemo.tsx]: Remove `export default`

**Pull request merged**